src/components/activity-groups-list/activity-group-item/activity-item/unknown.tsx

```diff
@@ -0,0 +1,120 @@
+import { View, Text } from 'react-native';
+
+import { Divider } from 'src/components/divider/divider';
+import { ExternalLinkButton } from 'src/components/icon/external-link-button/external-link-button';
+import { PublicKeyHashText } from 'src/components/public-key-hash-text/public-key-hash-text';
+import { useNonZeroAmounts } from 'src/hooks/use-non-zero-amounts.hook';
+import { ActivityAmount } from 'src/interfaces/non-zero-amounts.interface';
+import { useSelectedRpcUrlSelector } from 'src/store/settings/settings-selectors';
+import { formatSize } from 'src/styles/format-size';
+import { calculateDollarValue } from 'src/utils/activity.utils';
+import { isDefined } from 'src/utils/is-defined';
+import { tzktUrl } from 'src/utils/linking.util';
+
+import { ActivityGroupAmountChange } from '../activity-group-amount-change/activity-group-amount-change';
+import { ActivityGroupDollarAmountChange } from '../activity-group-dollar-amount-change/activity-group-dollar-amount-change';
+import {
+  useActivityCommonStyles,
+  useActivityDetailsStyles,
+  useActivityGroupItemStyles
+} from '../activity-group-item.styles';
+import { ItemAmountChange } from '../item-amount-change/item-amount-change';
+import { ActivityGroupItemSelectors } from '../selectors';
+import { AbstractItem } from './abstract-item';
+import { ActivityItemProps } from './item.props';
+
+export const Unknown: FC<ActivityItemProps> = ({ activity }) => {
+  const nonZeroAmounts = useNonZeroAmounts(activity.tokensDeltas);
+
+  return (
+    <AbstractItem
+      status={activity.status}
+      timestamp={activity.timestamp}
+      face={<Face nonZeroAmounts={nonZeroAmounts} />}
+      details={<Details hash={activity.hash} nonZeroAmounts={nonZeroAmounts} />}
+    />
+  );
+};
+
+const Face: FC<{ nonZeroAmounts: Array<ActivityAmount> }> = ({ nonZeroAmounts }) => {
+  const styles = useActivityGroupItemStyles();
+  const commonStyles = useActivityCommonStyles();
+
+  return (
+    <View style={[commonStyles.row, commonStyles.itemsCenter]}>
+      <View style={styles.flex}>
+        <View style={[commonStyles.row, commonStyles.justifyBetween, commonStyles.itemsStart]}>
+          <Text style={styles.oprationTitle}>Interaction</Text>
+          <ActivityGroupAmountChange nonZeroAmounts={nonZeroAmounts} />
+        </View>
+        <View style={[commonStyles.row, commonStyles.justifyBetween, commonStyles.itemsStart]}>
+          <Text style={styles.oprationSubtitle}>-</Text>
+          <ActivityGroupDollarAmountChange dollarValue={calculateDollarValue(nonZeroAmounts)} />
+        </View>
+      </View>
+    </View>
+  );
+};
+
+const Details: FC<{ hash: string; nonZeroAmounts: Array<ActivityAmount> }> = ({ hash, nonZeroAmounts }) => {
+  const selectedRpcUrl = useSelectedRpcUrlSelector();
+  const styles = useActivityDetailsStyles();
+
+  const commonStyles = useActivityCommonStyles();
+
+  const { positiveAmounts, negativeAmounts } = useMemo(() => {
```

Comment:
Sorry, could you move this logic into a separate function? The same is there: https://github.com/madfish-solutions/templewallet-mobile/pull/936/files#diff-628bc4d2ac2def1b6db1dab9e7e04f77843c1b3cf2f90872c204b5dd936242f1R75